core/src/main/java/com/yahoo/oak/KeyUtils.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2020, Verizon Media.
+ * Licensed under the terms of the Apache 2.0 license.
+ * Please see LICENSE file in the project root for terms.
+ */
+
+package com.yahoo.oak;
+
+import com.yahoo.oak.ValueUtils.ValueResult;
+
+class KeyUtils {
+
+
+    /* ==================== Methods for operating on existing off-heap keys ==================== */
+    
+    
+    public static <K> int compareKeyAndSerializedKey(K key, OakScopedReadBuffer serializedKey, OakComparator<K> cmp) 
+            throws DeletedMemoryAccessException {
+        if (((ScopedReadBuffer) serializedKey).s.lockRead() != ValueResult.TRUE) {
```

Comment:
It is a bit disturbing that lock/unlock methods were "overloaded" for Nova start/stop. Can we use some other API name?

Reply:
If we used something else, then Nova would not be a simple plugin 
we need to replace/duplicate everything that uses lock to the new API